presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeAsyncDataCacheCleanupAPI.java

```diff
@@ -0,0 +1,203 @@
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.nativeworker.NativeApiEndpointUtils.fetchScalarLongMetrics;
+import static com.facebook.presto.nativeworker.NativeApiEndpointUtils.sendWorkerRequest;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createCustomer;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+public class TestPrestoNativeAsyncDataCacheCleanupAPI
+        extends AbstractTestQueryFramework
+{
+    private static final String memoryCacheCleanupEndPoint = "/v1/operation/server/clearCache?type=memory";
+    private static final String ssdCacheCleanupEndPoint = "/v1/operation/server/clearCache?type=ssd";
+    private static final String metricsEndPoint = "/v1/info/metrics";
+    private static final String writeToSsdEndPoint = "/v1/operation/server/writeSsd";
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createQueryRunner(true, false, true, true);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createJavaQueryRunner();
+    }
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createCustomer(queryRunner);
+    }
+
+    @Test
+    public void testAsyncDataCacheCleanup() throws Exception
+    {
+        Session session = Session.builder(super.getSession())
+                .setCatalogSessionProperty("hive", "node_selection_strategy", "SOFT_AFFINITY")
+                .build();
+
+        QueryRunner queryRunner = getQueryRunner();
+        DistributedQueryRunner distributedQueryRunner = (DistributedQueryRunner) queryRunner;
+        Set<InternalNode> workerNodes = getWorkerNodes(distributedQueryRunner);
+
+        // 1. Collect initial cache metrics
+        Metrics initialMetrics = collectCacheMetrics(workerNodes, metricsEndPoint);
+        assertEquals(0, initialMetrics.memoryCacheHits, "Cache hits should be zero initially.");
+        assertEquals(0, initialMetrics.memoryCacheEntries, "Cache entries should be zero initially.");
+
+        // 2. Execute queries to populate cache
+        for (int i = 0; i < 4; i++) {
+            queryRunner.execute(session, "SELECT count(*) FROM customer");
+        }
+        TimeUnit.SECONDS.sleep(60); // Sleep to allow cache updates
```

Comment:
Sleeping 60 seconds? Is it too long? This means this test would take at least 5 minutes.